### PR TITLE
Changes profile and groups calls to use JSON encoding

### PR DESCRIPTION
**Pull request by Ruenzuo**
It seems that some POST and PUT calls are more reliable if we use JSON encoding for the request bodies. This PR includes that and unit testing fixes.

### CODE REVIEW

File: Example/ExampleTests/XNGProfileEditingTests.m [hunk 1]
```diff
@@ -102,23 +99,23 @@ - (void)testUpdateUsersPrivateAddress {
         expect(request.URL.host).to.equal(@"api.xing.com");
         expect(request.URL.path).to.equal(@"/v1/users/me/private_address");
         expect(request.HTTPMethod).to.equal(@"PUT");
-
+
```
Comment (pietbrauer) [inline Example/ExampleTests/XNGProfileEditingTests.m]: That should probably not be in here.

**Comment by pietbrauer** [comment]
Fine with me, if the whitespace is out.

File: Example/ExampleTests/XNGProfileEditingTests.m (hunk 1 above)
Comment (Ruenzuo) [inline Example/ExampleTests/XNGProfileEditingTests.m]: No idea how that ended up there ¯\_(ツ)_/¯

**Comment by Ruenzuo** [comment]
Cool, thank you @pietbrauer! :blush:

**Pull request merged**